static/app/views/ddm/metricScratchpad.tsx

```diff
@@ -0,0 +1,266 @@
+    default: state.default,
+    selected,
+    add,
+    update,
+    remove,
+    select,
+    setDefault,
+  };
+}
+
+export function ScratchpadSelector() {
+  const scratchpads = useScratchpads();
+  const router = useRouter();
+  const query = router.location.query ?? {};
+  // select the default scratchpad when the component mounts
+  useEffect(() => {
+    if (scratchpads.default && !scratchpads.selected && !query.widgets) {
+      scratchpads.select(scratchpads.default);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // saves all changes to the selected scratchpad to local storage
+  useEffect(() => {
+    if (scratchpads.selected) {
+      scratchpads.update(scratchpads.selected, query);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [query]);
+
+  // changes the query when a scratchpad is selected, clears it when none is selected
+  useEffect(() => {
+    if (scratchpads.selected) {
+      const selectedQuery = scratchpads.all[scratchpads.selected].query;
+      updateQuery(router, selectedQuery);
+    } else if (scratchpads.selected === null) {
+      clearQuery(router);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [scratchpads.selected]);
+
+  const isDefaultSelected = scratchpads.default === scratchpads.selected;
+
+  return (
+    <ScratchpadGroup>
+      <Button
+        size="sm"
+        disabled={!scratchpads.selected}
+        onClick={() => {
+          if (isDefaultSelected) {
+            scratchpads.setDefault(null);
+          } else {
+            scratchpads.setDefault(scratchpads.selected ?? null);
+          }
+        }}
+        icon={<IconBookmark isSolid={isDefaultSelected} />}
+      >
+        {isDefaultSelected ? t('Remove default') : t('Set as default')}
+      </Button>
+      <SaveAsDropdown
+        onSave={name => {
+          scratchpads.add(name, query);
+        }}
+        mode={scratchpads.selected ? 'fork' : 'save'}
+      />
+      <CompactSelect
+        options={Object.values(scratchpads.all).map((s: any) => ({
+          value: s.id,
+          label: s.name,
+          trailingItems: (
+            <Fragment>
+              <StyledDropdownIcon>
+                <IconDelete
+                  onClick={() => {
+                    openConfirmModal({
+                      onConfirm: () => scratchpads.remove(s.id),
+                      message: t('Are you sure you want to delete this scratchpad?'),
+
+                      confirmText: t('Delete'),
+                    });
+                  }}
+                />
+              </StyledDropdownIcon>
+            </Fragment>
+          ),
+        }))}
+        value={scratchpads.selected ?? ''}
+        closeOnSelect={false}
+        onChange={option => {
+          if (option.value === scratchpads.selected) {
+            scratchpads.select(null);
+          } else {
+            scratchpads.select(option.value);
+          }
+        }}
```

Comment:
This duplicates the setting default logic. You could have one `toggleDefault` function for that.
```ts
function toggleDefault(value?: string) {
  if (value === scratchpads.selected) {
    scratchpads.select(null);
  } else {
    scratchpads.select(value ?? null);
  }
} 
```